syft/pkg/dpkg_metadata_test.go

```diff
@@ -12,39 +12,52 @@ import (
 
 func TestDpkgMetadata_pURL(t *testing.T) {
 	tests := []struct {
-		distro   linux.Release
+		name     string
+		distro   *linux.Release
 		metadata DpkgMetadata
 		expected string
 	}{
 		{
-			distro: linux.Release{
-				ID: "debian",
+			name: "go case",
+			distro: &linux.Release{
+				ID:        "debian",
+				VersionID: "11",
 			},
 			metadata: DpkgMetadata{
-				Package:      "p",
-				Source:       "s",
-				Version:      "v",
-				Architecture: "a",
+				Package: "p",
+				Source:  "s",
+				Version: "v",
 			},
-			expected: "pkg:deb/debian/p@v?arch=a",
+			expected: "pkg:deb/debian/p@v?distro=debian-11",
 		},
 		{
-			distro: linux.Release{
-				ID: "ubuntu",
+			name: "with arch info",
+			distro: &linux.Release{
+				ID:        "ubuntu",
+				VersionID: "16.04",
 			},
 			metadata: DpkgMetadata{
 				Package:      "p",
 				Source:       "s",
 				Version:      "v",
 				Architecture: "a",
 			},
-			expected: "pkg:deb/ubuntu/p@v?arch=a",
+			expected: "pkg:deb/ubuntu/p@v?arch=a&distro=ubuntu-16.04",
```

Comment:
One question, though: should these be more like `pkg:deb/ubuntu/p@v?arch=a&distro=16.04` without the `ubuntu-`? (After looking again at: https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#deb )

Reply:
Honestly, I'm not certain. The spec only has one "good" example with this here: https://github.com/package-url/purl-spec/blob/1b1e9b2afa6de1c855225fb08cb46878ad653925/PURL-SPECIFICATION.rst#some-purl-examples
```
pkg:deb/debian/curl@7.50.3-1?arch=i386&distro=jessie
pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
```

The debian distro code-name is a bad idea for us, so I didn't approach it that way... so the fedora example was used as a basis. That being said, there is essentially 0 verbiage on correct usage of `distro` even though it is a suggested qualifier (that we need). 

References:
- https://github.com/package-url/purl-spec/tree/a21a9ecbab171fe06d1ac88843bc3721a42dbed0#known-qualifiers-keyvalue-pairs
- https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#rpm

Since there is little direction from the spec I'm open to other options here.

Reply:
From an offline conversation: we'll leave this as is for now since it seems more semantically correct to the term "distro" but are open to changing in the future.